dattri/algorithm/influence_function.py

```diff
@@ -5,15 +5,23 @@
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from typing import List, Tuple
+    from collections.abc import Callable
+    from typing import Any, Dict, List, Optional, Tuple, Union
 
     from torch import Tensor
+    from torch.utils.data import DataLoader
 
+    from dattri.task import AttributionTask
+
+import warnings
 from functools import partial
 
 import torch
+from torch.func import grad
```

Comment:
How about just use torch.func.grad when needed? Since vmap is used in this way.

Reply:
Fixed.